### ISSUE TITLE
`in-package` not working for common lisp

### ISSUE THREAD

**Issue by glyh** [Olical/conjure#310]
As title. Again, I'm using this [script](https://gist.github.com/glyh/c7bcbae7961ee52fa0c21c30a447650c) to start swank, so it might be possible the configuration on my side is bugged.

To reproduce, evaluate the following sentences seperately:
```lisp
(defpackage :test
  (:use :cl))

(in-package :test)

*package*
```
This would yield:
```lisp
; Sponsored by @martinklepsch ❤
; --------------------------------------------------------------------------------
; 127.0.0.1:4005 (connected)
; --------------------------------------------------------------------------------
; eval (selection): (defpackage :test...
; --------------------------------------------------------------------------------
; eval (current-form): (in-package :test)
#<PACKAGE "TEST">
; --------------------------------------------------------------------------------
; eval (selection): *package*
#<PACKAGE "COMMON-LISP-USER">

```
with swank log:
```
;; Swank started at port: 4005.
decode-message
decode-message
dispatch-event: (:EMACS-REX (SWANK:EVAL-AND-GRAB-OUTPUT "(defpackage :test
  (:use :cl))
") "cl-user" T 1)
send-event: #<SB-THREAD:THREAD "worker" RUNNING {100395FC53}> (:EMACS-REX (SWANK:EVAL-AND-GRAB-OUTPUT "(defpackage :test
  (:use :cl))
") "cl-user" 1)
wait-for-event: (:EMACS-REX . SWANK::_) NIL
dispatch-event: (:NEW-FEATURES (CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :SWANK :ROS.SCRIPT :QUICKLISP :QUICKLISP-SUPPORT-HTTPS :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :ROS.INIT :X86-64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL :UNIX))
encode-message
dispatch-event: (:RETURN #<SB-THREAD:THREAD "worker" FINISHED values: T {100395FC53}> (:OK ("" "#<PACKAGE \"TEST\">")) 1)
encode-message
decode-message
dispatch-event: (:EMACS-REX (SWANK:EVAL-AND-GRAB-OUTPUT "(in-package :test)") "cl-user" T 1)
send-event: #<SB-THREAD:THREAD "worker" RUNNING {10039669E3}> (:EMACS-REX (SWANK:EVAL-AND-GRAB-OUTPUT "(in-package :test)") "cl-user" 1)
wait-for-event: (:EMACS-REX . SWANK::_) NIL
dispatch-event: (:RETURN #<SB-THREAD:THREAD "worker" FINISHED values: T {10039669E3}> (:OK ("" "#<PACKAGE \"TEST\">")) 1)
encode-message
decode-message
dispatch-event: (:EMACS-REX (SWANK:EVAL-AND-GRAB-OUTPUT "*package*
") "cl-user" T 1)
send-event: #<SB-THREAD:THREAD "worker" RUNNING {1003978E13}> (:EMACS-REX (SWANK:EVAL-AND-GRAB-OUTPUT "*package*
") "cl-user" 1)
wait-for-event: (:EMACS-REX . SWANK::_) NIL
dispatch-event: (:RETURN #<SB-THREAD:THREAD "worker" FINISHED values: T {1003978E13}> (:OK ("" "#<PACKAGE \"COMMON-LISP-USER\">")) 1)
encode-message
```

**Comment by glyh** [comment]
I've found something that may help: 
https://github.com/kovisoft/slimv/blob/master/ftplugin/swank.py

**Comment by glyh** [comment]
OK, seems like the namespace is hard-coded into the client.

**Comment by Olical** [comment]
I'm afraid I don't really have personal experience with this. Should the namespace be something that's only set once as a default or not set at all by the client? Can I parse the namespace out of the buffer / filename and send that with your evaluations?

I'm wondering if it works like Clojure's namespaces where I can read it from the file header and send it with each eval.

**Comment by glyh** [comment]
Well structured code in a file should set in-package only once, and the rest of the code in that file all works under that package.

**Comment by Olical** [comment]
I'm looking at this right now (I know, huge delay 😢 ) but Conjure never sets or resets the package name. I was looking into sending `in-package` to the REPL every time you eval using the name found in `defpackage` like I do with Clojure but that doesn't seem to help. It also seems to cause weird issues with other evals.

**Comment by Olical** [comment]
I _think_ I've improved this? CL support still has holes in it and needs more love, but I think things should be a little better now, I'm trying to evaluate within the context of your `defpackage` expression but I'm not sure if it's working.

This is made more difficult because I can't find docs or source code for `swank:eval-and-grab-output` which seems to be the crux of evaluating CL over swank. There's an argument that I _think_ might be the package name but it could just be a debug label for all I know. Hopefully this fixes it though! It seemed to work for me and give me contextual evals but again, I don't know if I'm doing it wrong since I'm not experienced with CL.

I also added a little duct tape to work around the "first eval does nothing" bug... I just send `:ok` to it on connection which should be a noop most of the time but seems to wake the server up nicely and make the second eval work just fine.
